### DISCUSSION TITLE
duckdb::StructTypeInfo is not available in the amalgamated header file

### DISCUSSION THREAD

**Discussion by mbasmanova** [duckdb/duckdb#4839] [General]
I'm trying to convert DuckDb's LogicalType representing a struct to Velox type RowType. I'm seeing that LogicalType::AuxInfo() returns a pointer to StructTypeInfo, but I cannot cast to it because it is not defined in the amalgamated header file. I wonder if it would be possible to expose StructTypeInfo in the header file.

CC: @Mytherin

**Comment by Mytherin** [comment]
The `AuxInfo` is an internal type that should not be accessed. You can use the `StructType` static methods to operate on struct types:

```cpp
struct StructType {
	DUCKDB_API static const child_list_t<LogicalType> &GetChildTypes(const LogicalType &type);
	DUCKDB_API static const LogicalType &GetChildType(const LogicalType &type, idx_t index);
	DUCKDB_API static const string &GetChildName(const LogicalType &type, idx_t index);
	DUCKDB_API static idx_t GetChildCount(const LogicalType &type);
};
```

**Reply by mbasmanova** [reply]
@Mytherin Mark, thank you for a quick reply. I didn't realize I can StructType::GetChildTypes(type) and similar. Let me try this.
